lib/init/grass.py

```diff
@@ -2404,10 +2442,16 @@ def main():
                                         mapset_settings.location,
                                         mapset_settings.mapset,
                                         grass_env_file)
+        elif sh in ['zsh']:
+            shell_process = shell_startup(mapset_settings.full_mapset,
+                                          mapset_settings.location,
+                                          grass_env_file,
+                                          "zsh")
         elif sh in ['bash', 'msh', 'cygwin']:
-            shell_process = bash_startup(mapset_settings.full_mapset,
-                                         mapset_settings.location,
-                                         grass_env_file)
+            shell_process = shell_startup(mapset_settings.full_mapset,
+                                          mapset_settings.location,
+                                          grass_env_file,
+                                          "bash")
```

Comment:
May I suggest to combine these two `elif`s:
```bash
else if sh in ['bash', 'zsh', 'msh', 'cygwin']:
   shell_process = ... grass_env_file, sh)
```
and handle `zsh` and the other `sh`'s inside the startup function.

Reply:
Sounds good to me.

Reply:
Per @wenzeslaus suggestion, let us leave this for another PR.
I have a feeling this might be totally reworked anyway for GRASS 8 if we attempt to remove the HOME hack.